x/interchainstaking/keeper/address_map.go

```diff
@@ -15,6 +15,30 @@ func (k *Keeper) GetRemoteAddressMap(ctx sdk.Context, localAddress []byte, chain
 	return value, value != nil
 }
 
+// IterateUserMappedAccounts iterates over all the user mapped accounts.
+func (k Keeper) IterateUserMappedAccounts(ctx sdk.Context, localAddress []byte, fn func(index int64, chainID string, remoteAddressBytes []byte) (stop bool)) {
+	// noop
+	if fn == nil {
+		return
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.GetRemoteAddressPrefix(localAddress))
+	defer iterator.Close()
+
+	i := int64(0)
+	for ; iterator.Valid(); iterator.Next() {
+		value := iterator.Value()
+		key := iterator.Key()
+		chainIDBytes := key[len(types.GetRemoteAddressPrefix(localAddress)):]
```

Comment:
Does the iterator return the entire key, or just the bit appended to the prefix. I feel it may be the latter?

Reply:
it returns complete key with prefix